addons/website_sale/models/website_snippet_filter.py

```diff
@@ -37,3 +42,88 @@ def _get_hardcoded_sample(self, model):
                 # merge definitions
             samples = merged
         return samples
+
+    def _filter_records_to_values(self, records, is_sample=False):
+        res_products = super()._filter_records_to_values(records, is_sample)
+        if self._get_model()._name == 'product.product':
+            max_nb_chars = 100
+            for res_product in res_products:
+                product = res_product.get('_record')
+                if not is_sample:
+                    res_product.update(product._get_combination_info_variant())
+                if res_product['description_sale'] and len(res_product['description_sale']) > max_nb_chars:
+                    res_product['description_sale'] = "%s ..." % res_product['description_sale'][:max_nb_chars]
+        return res_products
+
+    def _get_products(self, website, product_selection, limit, context):
+        handler = getattr(self, '_get_products_%s' % product_selection, self._get_products_latest_sold)
+        search_domain = context.get('search_domain')
+        domain = [('website_published', '=', True)] + website.website_domain() + (search_domain or [])
+        products = handler(website, limit, domain, context)
+        return self._filter_records_to_values(products, False)
+
+    def _get_products_latest_sold(self, website, limit, domain, context):
+        products = []
+        sale_orders = self.env['sale.order'].sudo().search([
+            ('state', 'in', ('sale', 'done'))
+        ], limit=8, order='date_order DESC')
+        if sale_orders:
+            products = []
+            for sale_order in sale_orders:
+                products.extend(sale_order.order_line.product_id.ids)
+            products_ids = [id for id, _ in Counter(products).most_common()]
+            if products_ids:
+                domain.append(['id', 'in', products_ids])
+                products = self.env['product.product'].with_context(display_default_code=False).search(domain)
```

Comment:
limit here instead of return

Reply:
This is intentional: if the limit is applied in the search, records will be excluded before being sorted

Reply:
remember me why not a browse ? 
If not need to reapply ACL, you already have the id so browse(products[:limit)) will preserve order